products/microsoft_defender_for_endpoints.py

```diff
@@ -0,0 +1,114 @@
+import csv
+import json
+import os
+import time
+import datetime
+from pprint import pprint
+import click
+
+import urllib.request
+import urllib.parse
+import sys
+import requests
+
+
+def build_query(filters):
+    query_base = ''
+
+    for key, value in filters.items():
+        if key == 'days':
+            query_base += f'| where Timestamp > ago({value}d)'
+
+        if key == 'minutes':
+            query_base += f'| where Timestamp > ago({value}m)'
```

Comment:
Can both of these keys be provided at the same time? It could produce some confusing results where the `minutes` is prioritized over the `days`.

Reply:
They should be able to be used in tandem along with ```hostname``` and ```username```. Since it's running a for loop over all the filter items if both days and minutes are used together it _should_ allow for both. Does Defender not allow days and minutes be used together?

Reply:
Logically, a query that includes `| where Timestamp > ago(15m) | where Timestamp > ago(7d)` will go through, but it will ignore the less strict portion of the query in the results (e.g. all the results will be from the last 15 minutes). That might be fine from a design standpoint for surveyor, but I thought it worth mentioning.